examples/sol-anchor-contract/programs/sol-anchor-contract/src/lib.rs

```diff
@@ -62,10 +43,21 @@ pub mod example_sol_anchor_contract {
         let mut loan_max_value = loan_max_price
             .checked_mul(loan_qty)
             .ok_or(ErrorCode::Overflow)?;
+
+        // lets get the maximum loan value based on computation
+        // i.e {} * 10^({})
+        // loan_max_value * 10^(loan_price.expo)
+        let base: i32 = 10;
+        let exponent: i32 = loan_price.expo;
+        let result = (base as f64).powi(exponent.abs());
```

Comment:
I think you can do `let result = (base as f64).powi(exponent);`

Reply:
Please note that code line "let result = if exponent < 0 { 1.0 / result } else { result };" is dependent on ".abs" being used on code line "let result = (base as f64).powi(exponent.abs());". 

Since we are getting absolute value of exponent and using it in the computation on code line "52", we check if exponent was negative or positive in the computation done on code line "let result = if exponent < 0 { 1.0 / result } else { result };"

I tested the code after removing ".abs()" and both "result_loan_value" and "result_collateral_value" had value zero. The two variables had postitive values if we maintain ".abs" during our computation i.e

Program log: The maximum loan value is 186894000000 * 10^(-8) = 1868.94.
Program log: The minimum collateral value is 300045300000 * 10^(-8) = 3000.453.